backend/app/user_handler.go

```diff
@@ -167,76 +171,27 @@ func (h *Handler) RegisterHandler(c *gin.Context) {
 		}
 	}
 
-	code := internal.GenerateRandomCode()
-	subject, body := h.mailService.SignUpMailContent(code, h.config.MailSender.Timeout, request.Name, h.config.Server.Host)
-
-	err := h.mailService.SendMail(h.config.MailSender.Email, request.Email, subject, body)
-	if err != nil {
-		log.Error().Err(err).Msg("failed to send verification code")
-		InternalServerError(c)
-		return
-	}
-
-	// hash password
-	hashedPassword, err := internal.HashAndSaltPassword([]byte(request.Password))
+	wf, err := h.ewfEngine.NewWorkflow("user-registration")
 	if err != nil {
-		log.Error().Err(err).Msg("error hashing password")
+		log.Error().Err(err).Msg("failed to start registration workflow")
 		InternalServerError(c)
 		return
 	}
 
-	isAdmin := internal.Contains(h.config.Admins, request.Email)
-
-	user := models.User{
-		Username: request.Name,
-		Email:    request.Email,
-		Password: hashedPassword,
-		Code:     code,
-		Verified: existingUser.Verified,
-		Admin:    isAdmin,
-	}
-
-	if getErr != gorm.ErrRecordNotFound {
-		if !user.Verified {
-			user.ID = existingUser.ID
-			err = h.db.UpdateUserByID(&user)
-			if err != nil {
-				log.Error().Err(err).Send()
-				InternalServerError(c)
-				return
-			}
-		}
+	wf.State = ewf.State{
+		"name":     request.Name,
+		"email":    request.Email,
+		"password": request.Password,
 	}
 
-	// create user model in db
-	if getErr != nil {
-		mnemonic, _, err := internal.SetupUserOnTFChain(h.substrateClient, h.config)
-		if err != nil {
-			log.Error().Err(err).Msg("failed to setup user on TFChain")
-			InternalServerError(c)
-			return
-		}
-		customer, err := internal.CreateStripeCustomer(request.Name, request.Email)
-		if err != nil {
-			log.Error().Err(err).Send()
-			InternalServerError(c)
-			return
-		}
-
-		user.Mnemonic = mnemonic
-		user.StripeCustomerID = customer.ID
+	h.ewfEngine.RunAsync(context.Background(), wf)
 
-		err = h.db.RegisterUser(&user)
-		if err != nil {
-			log.Error().Err(err).Send()
-			InternalServerError(c)
-			return
-		}
-	}
+	// send notification that user registered
```

Comment:
is it TODO: ?

Reply:
I removed it, it will be handled as a future work after this is merged